lib/parse.js

```diff
@@ -74,44 +81,60 @@ function parse (xml) {
  */
 
 function parsePlistXML (node) {
-  var i, new_obj, key, val, new_arr, res, d;
+  var i, new_obj, key, val, new_arr, res, counter, type;
 
   if (!node)
     return null;
 
   if (node.nodeName === 'plist') {
     new_arr = [];
-    for (i=0; i < node.childNodes.length; i++) {
-      // ignore comment nodes (text)
-      if (!shouldIgnoreNode(node.childNodes[i])) {
-        new_arr.push( parsePlistXML(node.childNodes[i]));
+    if (!isEmptyNode(node)) {
```

Comment:
I'd prefer to return early rather than wrap the whole thing in an `if` block. results in less nesting. e.g.,

```
if(isEmptyNode(node)) {
  return [];
}
```

It also eliminates the "not" modifier (`!`)  which makes the if condition ever-so-slightly easier to read.